### PR TITLE
Add support for symfony 4.2 (and some improvements)

### PR DESCRIPTION
**Pull request by AndreasA**
Added support for symfony 4.2 and also remove the whole symfony framework from requirements but use individual components (probably too many but still better than just everything).

### CODE REVIEW

**Comment by ywarnier** [comment]
Hi @AndreasA, thank you, but I must ask: are you sure you want to send that to our project? I get the original one is in read-only, but we don't think we'll keep the maintenance going for a very long time, so we'll look at this but it's likely this might have a better future in another fork (not sure, we also forked it after looking for an alternative).

**Comment by AndreasA** [comment]
Well. There currently is no one else who is maintaining it. So for now it would be nice to be in your but I guess if we keep using it we might fork again.
I am actually not sure why Sylius abandoned it, I think it is quite a useful bundle.

**Comment by ywarnier** [comment]
Maybe @jmontoyaa has some idea on the "why"?

**Comment by AndreasA** [comment]
Probably nobody to maintain it :) So shall I close the PR or do you want to merge it, it is finished. Have it working this way with symfony 4.2

**Comment by ywarnier** [comment]
@jmontoyaa will check that and get back to you tomorrow. Please leave it open for now, maybe he has other thoughts on that.

**Comment by jmontoyaa** [comment]
@ywarnier we are using this bundle in v2, so we are maintain it (at least for bugfixes)

File: composer.json [hunk 1]
```diff
@@ -24,7 +24,38 @@
...
+        "symfony/config": "^3.4|^4.1",
+        "symfony/console": "^3.4|^4.1",
+        "symfony/dependency-injection": "^3.4|^4.1",
+        "symfony/doctrine-bridge": "^3.4|^4.1",
+        "symfony/event-dispatcher": "^3.4|^4.1",
+        "symfony/expression-language": "^3.4|^4.1",
+        "symfony/filesystem": "^3.4|^4.1",
+        "symfony/finder": "^3.4|^4.1",
+        "symfony/form": "^3.4|^4.1",
+        "symfony/framework-bundle": "^3.4|^4.1",
```
Comment (jmontoyaa) [inline composer.json:31]: Why adding all this components? It won't be enough with only adding this?

```suggestion
        "symfony/framework-bundle": "^3.4|^4.1",
```

Comment (AndreasA) [reply composer.json:31]: Ah sorry, I actually copied too much. will adjust it and let you know

Comment (AndreasA) [reply composer.json:31]: List minified. However, for instance the options-resolver isn't part of the framework bundle.

**Comment by AndreasA** [comment]
Btw. don't forget that if you update your project, you need to update your classes that extend `Sylius\Bundle\SettingsBundle\Schema\SchemaInterface` that define your settings as I had to modify the `buildSettings` method to avoid using OptionsResolverInterface which doesn't exist anymore since Symfony 3.x.

**Comment by AndreasA** [comment]
@jmontoyaa Btw. can you force-update the latest commit hash on packagist for the package?

**Comment by jmontoyaa** [comment]
@AndreasA I just updated packagist, now it should be updated after every push.

**Comment by AndreasA** [comment]
@jmontoyaa Thanks

**Pull request merged**